master/internal/rm/kubernetesrm/informer_intg_test.go

```diff
@@ -213,12 +180,165 @@ func TestNodeInformer(t *testing.T) {
 					Object: node,
 				}
 			}
+
+			wg.Wait()
+			assert.Equal(t, tt.output, currNodes)
+		})
+	}
+}
+
+func TestEventListener(t *testing.T) {
+	cases := []struct {
+		name       string
+		expected   error
+		eventNames []string
+	}{
+		{"zero events", nil, []string{}},
+		{"listener success", nil, []string{"A"}},
+		{
+			"listener success & event ordering success",
+			nil,
+			[]string{"A", "B", "C", "D", "E"},
+		},
+		{
+			"listener success & event ordering failure",
+			nil,
+			[]string{"A", "B", "C", "D", "E"},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			var wg sync.WaitGroup
+			wg.Add(len(tt.eventNames))
+
+			ctx := context.TODO()
+			eventChan := make(chan watch.Event)
+			ordering := make([]string, 0)
+
+			mockOptsList, mockOptsWatch := initializeMockOpts("event")
+			mockEventInterface := &mocks.EventInterface{}
+
+			mockEventInterface.On("List", ctx, mockOptsList).Return(
+				&k8sV1.EventList{
+					ListMeta: metaV1.ListMeta{
+						ResourceVersion: "1",
+					},
+				},
+				nil)
+			mockEventInterface.On("Watch", ctx, mockOptsWatch).Return(&mockWatcher{c: eventChan}, nil)
+
+			mockEventHandler := func(event *k8sV1.Event) {
+				t.Logf("received event %v", event)
+				ordering = append(ordering, event.Name)
+				wg.Done()
+			}
+
+			i, err := newEventListener(
+				ctx,
+				mockEventInterface,
+				namespace,
+				mockEventHandler)
+			if err != nil {
+				assert.Nil(t, i)
+				assert.Error(t, tt.expected, err)
+				return
+			}
+			assert.NotNil(t, i)
+			assert.Equal(t, tt.expected, err)
+
+			go i.run()
+			for _, name := range tt.eventNames {
+				event := &k8sV1.Event{
+					ObjectMeta: metaV1.ObjectMeta{
+						ResourceVersion: "1",
+						Name:            name,
+					},
+				}
+				eventChan <- watch.Event{
+					Type:   watch.Modified,
+					Object: event,
+				}
+			}
+
 			wg.Wait()
+			assert.Equal(t, tt.eventNames, ordering)
+		})
+	}
+}
+
+func TestPreemptionListener(t *testing.T) {
+	cases := []struct {
+		testName string
+		expected error
+		names    []string
+	}{
+		{"zero preemptions", nil, []string{}},
+		{"informer success", nil, []string{"abc"}},
+		{
+			"informer success & event ordering success",
+			nil,
+			[]string{"A", "B", "C", "D", "E"},
+		},
+		{
+			"informer success & event ordering failure",
+			nil,
+			[]string{"A", "B", "C", "D", "E"},
+		},
```

Comment:
same comment, i assume failure cases are just in progress though

Reply:
ditto